### PR TITLE
Change gpu_device.cc so that per process fraction uses total memory fraction, not available memory

### PR DESCRIPTION
**Pull request by MostRahmani**
This change is for making the memory allocation from the whole memory not only the remaining memory to achieve better configuration for multi processes

### CODE REVIEW

**Comment by mrphoenix13** [comment]
this change was according to this issue #2176  @vrv

File: tensorflow/core/common_runtime/gpu/gpu_device.cc [hunk 1]
```diff
@@ -604,7 +604,7 @@ LocalDevice* BaseGPUDeviceFactory::CreateGPUDevice(
       allocated_memory -= min_system_memory;
     }
   } else {
-    allocated_memory *= config_memory_fraction;
```
Comment (vrv) [inline tensorflow/core/common_runtime/gpu/gpu_device.cc]: For clarity (so it's clear that this path doesn't use available_memory at all), can you change 598 to be just 

    int64 allocated_memory;
    if (config_memory_fraction == 0) {
      allocated_memory = available_memory;
      ...
    } else {
      allocated_memory = total_memory * config_memory_fraction;
    }

Comment (mrphoenix13) [inline tensorflow/core/common_runtime/gpu/gpu_device.cc]: good one. It's done.

File: tensorflow/core/common_runtime/gpu/gpu_device.cc [hunk 2]
```diff
@@ -595,10 +595,11 @@ LocalDevice* BaseGPUDeviceFactory::CreateGPUDevice(
   int64 total_memory, available_memory;
   CHECK(se->DeviceMemoryUsage(&available_memory, &total_memory));
 
-  int64 allocated_memory = available_memory;
+  int64 allocated_memory;
   double config_memory_fraction =
       options.config.gpu_options().per_process_gpu_memory_fraction();
   if (config_memory_fraction == 0) {
+      allocated_memory = available_memory;
```
Comment (vrv) [inline tensorflow/core/common_runtime/gpu/gpu_device.cc]: can you align the indentation to be two space?

Comment (mrphoenix13) [inline tensorflow/core/common_runtime/gpu/gpu_device.cc]: sorry for that. It's done

**Comment by vrv** [comment]
@tensorflow-jenkins test this please

(Thanks!  This was the intention of the field, so hopefully people aren't expecting the current behavior).

**Comment by mrphoenix13** [comment]
Is Android Demo App failure related to this change?! it seems that it cant add jenkins group!
@vrv

**Comment by vrv** [comment]
Looks like infrastructure failure, you can ignore. I will merge tomorrow!

Cc @jendap @caisq

**Pull request merged**